File: Projekt2/Kod/planner-java/planner.java

public class planner {
	public static void main (String args[]) {
        // Read the data
        String holding = args[0];
        String world = args[1];
        String[] trees = args[2].split(";");

        // Print the data
		System.out.println("# Stupid Java planner!");
		System.out.println("# Holding: " + holding);
		System.out.println("# World: " + world);
        for (String t : trees) {
            System.out.println("# Tree: " + t);
        }

        // Print the plan
		System.out.println("This is a stupid move!");
        int stacknr = 0;
        for (int i = 0; i < world.length(); i++) {
            if (world.charAt(i) == ';') 
                stacknr++;
            else if (world.charAt(i) != ' ') 
                break;
        }
		System.out.println("pick " + stacknr);
		System.out.println("drop " + stacknr);
	}
}

File: Projekt2/Kod/planner-haskell/planner.hs

-- How to use: 
--   compile this file into planner.cgi:
--   ghc -o planner.cgi --make planner.hs

import Network.CGI
import Data.Maybe (fromMaybe, fromJust)
import Data.List (findIndex)
import Control.Monad (liftM)
import Planner

type Block = String
type World = [[Block]]
type Tree = String


cgiMain :: CGI CGIResult
cgiMain = do setHeader "Content-type" "text/plain"
             (holding, world, trees) <- cgiInput
             let plan = findPlan holding world trees
             output (unlines plan)


findPlan :: Block -> World -> [Tree] -> [String]
findPlan holding word trees = ["pick 1", "drop 8"]


cgiInput :: CGI (Block, World, [Tree])
cgiInput = do holding <- liftM (fromMaybe "") (getInput "holding")
              worldStr <- liftM (fromMaybe "") (getInput "world")
              let world = [split ',' stack | stack <- split ';' worldStr]
              treesStr <- liftM (fromMaybe "") (getInput "trees")
              let trees = split ';' treesStr
              return (holding, world, trees)


split :: Char -> String -> [String]
split delim str
    | rest == "" = if null token then [] else [token] 
    | otherwise  = token : split delim (tail rest)
    where (token, rest) = span (/=delim) str


main :: IO ()
main = runCGI (handleErrors cgiMain)
